homeworks/franktoth/3_conditionals_loops/rock_paper_scissors.py

```diff
@@ -0,0 +1,51 @@
+# Get player names and desired number of rounds
+player1_name = input("Player 1, please enter your name: ")
+player2_name = input("Player 2, please enter your name: ")
+
+while True:
+  try:
+    rounds = int(input("How many rounds would you like to play? (Must be an odd number): "))
+    if rounds % 2 == 0:
+      print("The number of rounds must be odd. Please try again.")
+    else:
+      break
+  except ValueError:
+    print("Please enter a valid number of rounds.")
+
+# Initialize player scores
+player1_score = 0
+player2_score = 0
+
+# Play the game
+for round_num in range(1, rounds + 1):
+  print(f"\nRound {round_num}")
+
+  while True:
+    player1_choice = input(f"{player1_name}, please choose rock, paper, or scissors: ").lower()
+    if player1_choice not in ["rock", "paper", "scissors"]:
+      print("Invalid choice. Please choose rock, paper, or scissors.")
+    else:
+      break
+
+  while True:
+    player2_choice = input(f"{player2_name}, please choose rock, paper, or scissors: ").lower()
+    if player2_choice not in ["rock", "paper", "scissors"]:
+      print("Invalid choice. Please choose rock, paper, or scissors.")
+    else:
+      break
+
+  # Determine the winner of the round
```

Comment:
Ez nagyon közel van a jóhoz, de ha mindketten ugyanazt választják, akkor automatikusan a 2. játékos kapja a pontot.
Vizsgáld meg az elején, hogy a player_1_choice nem e ugyanaz mint a player_2_choice. Ebben az esetben ne kapjon egyik se pontot.
Ebből következik, hogy kezelni kéne ezt az esetet is úgy, hogy az adott iterációról (körről) nem ugrunk a következőre amíg eltérő jelet nem mondanak/mutatnak.

Példa:
berakhatnád a 23. és a 30. soron kezdődő logikát egy function-be, valami ilyesmi:

```
def get_choice(player_name):
    while True:
        choice = input(f"{player_name}, please choose rock, paper, or scissors: ").lower()
        if choice not in ["rock", "paper", "scissors"]:
            print("Invalid choice. Please choose rock, paper, or scissors.")
        else:
            return choice
```

Ezután csinálhatnál egy while True loopot a for loopodon belül:


```
while True:
    player1_choice = get_choice(player1_name)
    player2_choice = get_choice(player2_name)

    if player1_choice == player2_choice:
        print("It's a tie! Both players chose the same sign. Please choose again.")
    else:
        if (player1_choice == "rock" and player2_choice == "scissors") or \
           (player1_choice == "paper" and player2_choice == "rock") or \
           (player1_choice == "scissors" and player2_choice == "paper"):
            print(f"{player1_name} wins this round!")
            player1_score += 1
        else:
            print(f"{player2_name} wins this round!")
            player2_score += 1
        break  # ezzel a break-kel lépsz ki a while True-ból, mert fixen volt egy győztes és nem ugyanazt mondták
```